src/background/websocket-manager.ts

```diff
@@ -0,0 +1,361 @@
+    await updateScriptHash(this.socket, network, scripthash, this.marinaStore);
+  }
+
+  async forceUpdateAccount(accountID: AccountID, network: NetworkString): Promise<void> {
+    const account = selectAccount(accountID)(this.marinaStore.getState());
+    if (!account) throw new Error(`Account ${accountID} not found`);
+    const id = await account.getWatchIdentity(network);
+    const addresses = await id.getAddresses();
+
+    const scriptsHashes = addresses
+      .map((a) => address.toOutputScript(a.confidentialAddress).toString('hex'))
+      .map(reverseAndHash);
+
+    // init the reducer state in order to link account to scripthash
+    for (const scripthash of scriptsHashes) {
+      try {
+        selectUtxosMapByScriptHash(network, scripthash)(this.marinaStore.getState());
+      } catch {
+        this.marinaStore.dispatch(addScriptHash(accountID, scripthash, network));
+      }
+    }
+
+    for (const scriptHash of scriptsHashes) {
+      await this.forceUpdate(scriptHash, network);
+    }
+  }
+}
+
+// utility function converting an output script hex to a scripthash (reversed sha256 of the script)
+// scripthash is used by electrum protocol to identify a script to watch
+export function reverseAndHash(script: string): string {
+  return crypto.sha256(Buffer.from(script, 'hex')).reverse().toString('hex');
+}
+
+// this function is triggered each time a new status is received for a scripthash
+// it fetches the new txs from history and updates the utxos and transactions state
+async function updateScriptHash(
+  ws: ElectrumWS,
+  network: NetworkString,
+  scriptHash: string,
+  store: Store
+): Promise<void> {
+  const historyResp = (await ws.request(
+    'blockchain.scripthash.get_history',
+    scriptHash
+  )) as History;
+  const historyDiff = selectHistoryDiff(historyResp)(store.getState());
+  const accountID = selectAccountIDByScriptHash(scriptHash, network)(store.getState());
+  const blindKeyGetter = getPrivateBlindKeyGetter(store, network);
+  const walletScripts = await getAllWalletScripts(store, network);
+
+  for (const { txID, height } of historyDiff.confirmedTxs) {
+    if (height <= 0) continue;
+    const blockHeaderHex = (await ws.request('blockchain.block.header', height)) as string;
+    const blockHeader = deserializeBlockHeader(blockHeaderHex);
+    const action = confirmTx(txID, blockHeader.timestamp, network);
+    store.dispatch(action);
+  }
+
+  // for each new tx which is not in the state, we fetch the transaction and unblind it
+  for (const { txID, height } of historyDiff.newTxs) {
+    if (selectIsKnownTx(txID)(store.getState())) continue;
+    try {
+      store.dispatch(pushUpdaterLoader());
+      const txFromElectrum = await fetchTx(ws, txID, height > 0 ? height : undefined);
+      const { unblindedTx } = await unblindTransaction(txFromElectrum, blindKeyGetter);
+      const displayTx = toDisplayTransaction(unblindedTx, walletScripts, networks[network]);
```

Comment:
shall this be done in the presentation layer when user opend the tx detail page?